eth/state_accessor.go

```diff
@@ -73,13 +74,16 @@ func (eth *Ethereum) hashState(ctx context.Context, block *types.Block, reexec u
 	// The state is both for reading and writing, or it's unavailable in disk,
 	// try to construct/recover the state over an ephemeral trie.Database for
 	// isolating the live one.
+	// Use ReferenceRoot for ephemeral trie.Databases so reference counting
+	// logic is the same as a live trie.Database.
+	tdbConfig := &triedb.Config{HashDB: &hashdb.Config{ReferenceRoot: true}}
```

Comment:
this is where this change gets a little weird, but
- This code doesn't really ever run because we have reexecute = 0 at the top
- I checked all instances that create a triedb (non-test) that could flow here, but it only seems it could be from the chain

Reply:
actually this is not needed because the only time statedb is passed into this fn (hashState) is the one that is returned from it (in traceChain api)
So it is safe to not consider this.